File: core/src/test/java/ml/glassmc/loader/test/shard/TestReference.java
package ml.glassmc.loader.test.shard;

import ml.glassmc.loader.Listener;
import ml.glassmc.loader.Reference;

import java.util.HashMap;
import java.util.Map;

public class TestReference implements Reference {

    @Override
    public Map<Class<?>, Class<? extends Listener>> getListeners() {
        return new HashMap<Class<?>, Class<? extends Listener>>() {
            {
                put(TestHook.class, TestListener.class);
            }
        };
    }

}

File: core/src/test/kotlin/ml/glassmc/loader/test/shard/TestReference.kt
package ml.glassmc.loader.test.shard

import ml.glassmc.loader.Listener
import ml.glassmc.loader.Reference
import java.util.HashMap

class TestReference : Reference {

    override val listeners: Map<Class<*>, Class<out Listener>>
        get() = object: HashMap<Class<*>, Class<out Listener>>() {
            init {
                put(TestHook::class.java, TestTestListener::class.java)
            }
        }

}